### PR TITLE
[Feature] create new function such as current_user

### PR DESCRIPTION
**Pull request by shshenhua**
## What type of PR is this：
- [ ] Feature

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/15382
https://github.com/StarRocks/starrocks/issues/15285

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by shshenhua** [comment]
spark and presto can execute
select current_user() , select current_user
select user() , select user

but starrocks only execute such as
select current_user()
select user()

so enhance functions like spark and presto， now we can execute
select current_user
select user

**Comment by shshenhua** [comment]
starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -2092,6 +2092,7 @@ nonReserved
     | BACKEND | BACKENDS | BACKUP | BEGIN | BITMAP_UNION | BOOLEAN | BROKER | BUCKETS | BUILTIN
     | CAST | CATALOG | CATALOGS | CEIL | CHAIN | CHARSET | CURRENT | COLLATION | COLUMNS | COMMENT | COMMIT | COMMITTED
     | COMPUTE | CONNECTION | CONNECTION_ID | CONSISTENT | COSTS | COUNT | CONFIG
+    | CURRENT_USER
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:2095]: Why do you add so many key word? You can do it in the analysis state

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:2095]: You can implement it as a normal function

**Review by imay — COMMENTED** [review]
You can refer the current_version function.

**Comment by shshenhua** [comment]
run starrocks_fe_unittest

**Comment by shshenhua** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by shshenhua** [comment]
I remove the key word

spark and presto can execute "select current_user"
so, sr also wants to execute "select current_user"
@imay

**Comment by Astralidea** [comment]
mysql also support this. I think you can backport to 2.3~2.5, not only 2.3

**Review by Astralidea — DISMISSED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.5

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-2.4

**Pull request merged**